ophyd/ophydobj.py

```diff
@@ -373,12 +373,13 @@ def __repr__(self):
         return '{}({})'.format(self.__class__.__name__, info)
 
     def _repr_info(self):
+        'Yields pairs of (key, value) to generate the object repr'
         if self.name is not None:
             yield ('name', self.name)
 
         if self._parent is not None:
             yield ('parent', self.parent.name)
 
-    def __copy__(self):
-        info = dict(self._repr_info())
-        return self.__class__(**info)
+    def __getnewargs_ex__(self):
+        kwargs = dict(self._repr_info())
+        return ((), kwargs)
```

Comment:
`__getnewargs__`, as I understand it, is a replacement for `__getstate__` ~or `__copy__`~. It indicates the init kwargs necessary to recreate a given object. Since the early versions of ophyd, `_repr_info` has been used to generate said kwargs, making this method fairly simple.

... however, as I went to prove the implementation with an example, I realize this change is not correct. I'm glad you brought attention to it! I'll re-add the `__copy__` method. For reference, the 2nd assertion will fail here:

```python
import copy


class Test:
    def __init__(self, **kw):
        self.kw = kw

    def __getnewargs_ex__(self):
        print('getnewargs called')
        return ((), self.kw.copy())


a = Test('a', a=1, b=2)
b = copy.copy(a)

assert id(a) != id(b)
assert id(a.kw) != id(b.kw)
```